src/dvc_task/proc/manager.py

```diff
@@ -142,17 +146,20 @@ def handle_closed_process():
                         handle_closed_process()
                         raise ProcessLookupError from exc
                 raise
+        return False
```

Comment:
I think it would be better to raise an exception in this case, it's not really obvious what the boolean return values indicate.

Maybe we should add a new exception type for dvc-task like

```
class ProcessClosedError:
    """Process has already exited."""
    def __init__(self, returncode):
        self.returncode = returncode
```

That way the caller can check for this exception and handle it as needed, and since `exc.returncode` will be available, the caller will also be able to see if the process succeeded (exited with 0) or failed (exited with nonzero)

And then here we would just raise `ProcessClosedError(process_info.returncode)` instead of returning false

Reply:
Thinking about this some more, the existing behavior was probably actually a bug, and we should have already been raising `ProcessLookupError` (or a dvc-task specific subclass of that exception) to be consistent with `os.kill()` signal behavior